v6_carrier_py/master.py

```diff
@@ -89,3 +92,15 @@ def column_names(client: ClientContainerProtocol, data, *args, exclude_orgs=(),
 
     # return all the messages from the nodes
     return column_set
+
+
+def correlation_matrix(client: ClientContainerProtocol, data, keys=None, *args, **kwargs):
+    """
+    Compute a correlation matrix over all datasets together. Data will be joined using the specified key.
```

Comment:
It would be nice to add a bit more documentation about what is done in the different edgecases  that you actually test in your tests.

Reply:
Done